### PR TITLE
[dualtor] Add test to simulate server reboot

### LINKED ISSUE #12039: [dualtor][test gap] PXEboot scenario
**Description**

**Steps to reproduce the issue:**
1. shutdown icmp_responder
2. reboot server

**Describe the results you expected:**
Scenario 1 link flap after boot up. 
One side ToR toggles mux, and  the other side is able to sync up in short amount of time. 
Scenario 2 start icmp_responder after boot up 
Both ToRs get back to healthy state immediately. 
 

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by opcoder0**
### Description of PR

Add test to simulate server reboot via restart of `icmp_responder` and verify that the TORs get back to `healthy` status after the restart is complete.

Summary:
Fixes #12039 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?

Address test gap.

#### How did you do it?

Add test to simulate server reboot via shutdown of `icmp_responder` service and verified that the TOR health is back to normal after reboot is complete.

#### How did you verify/test it?

Ran tests on dualtor-aa testbeds.

#### Any platform specific information?

None

#### Supported testbed topology if it's a new test case?

dualtor

### Documentation

None

### CODE REVIEW

File: tests/dualtor_mgmt/test_server_failure.py [hunk 1]
```diff
@@ -90,3 +92,42 @@ def lower_tor_mux_state_verfication(state, health):
...
+
+    def upper_tor_mux_state_verification(state, health):
+        mux_state_upper_tor = show_muxcable_status(upper_tor_host)
+        return (mux_state_upper_tor[test_iface]['status'] == state and
+                mux_state_upper_tor[test_iface]['health'] == health)
+
+    def lower_tor_mux_state_verfication(state, health):
+        mux_state_lower_tor = show_muxcable_status(lower_tor_host)
+        return (mux_state_lower_tor[test_iface]['status'] == state and
+                mux_state_lower_tor[test_iface]['health'] == health)
```
Comment (zjswhhh) [inline tests/dualtor_mgmt/test_server_failure.py:109]: can you try using dualtor test utilities? 

https://github.com/sonic-net/sonic-mgmt/blob/b6d492467abc69c3db32f381cae7ec0bab57ee34/tests/common/dualtor/control_plane_utils.py#L255-L261

**Review by zjswhhh — COMMENTED** [review]
Seems that your chance didn't cover scenario 1. 

The goal is to confirm when one ToR toggles mux direction, the other ToR will be in sync eventually even without icmp heartbeats.

```
Scenario 1 link flap after boot up.
One side ToR toggles mux, and the other side is able to sync up in short amount of time.
```

**Review by lolyu — APPROVED** [review]
LGTM

**Review by zjswhhh — APPROVED** [review]
lgtm

**Review by zjswhhh — APPROVED** [review]
lgtm

**Comment by zjswhhh** [comment]
Hi @opcoder0  - can you address the PR checker failure? 

```
dualtor_mgmt/test_server_failure.py::test_server_reboot[active-standby] 
-------------------------------- live log call ---------------------------------
17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet4" in full_dut_fanout_port_map
17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet8" in full_dut_fanout_port_map
17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet12" in full_dut_fanout_port_map
17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet16" in full_dut_fanout_port_map
17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet20" in full_dut_fanout_port_map
17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet24" in full_dut_fanout_port_map
17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet28" in full_dut_fanout_port_map
```

**Review by wangxin — DISMISSED** [review]

**Comment by opcoder0** [comment]
> Hi @opcoder0 - can you address the PR checker failure?
> 
> ```
> dualtor_mgmt/test_server_failure.py::test_server_reboot[active-standby] 
> -------------------------------- live log call ---------------------------------
> 17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet4" in full_dut_fanout_port_map
> 17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet8" in full_dut_fanout_port_map
> 17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet12" in full_dut_fanout_port_map
> 17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet16" in full_dut_fanout_port_map
> 17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet20" in full_dut_fanout_port_map
> 17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet24" in full_dut_fanout_port_map
> 17:01:08 dual_tor_utils._shutdown_fanout_tor_intf L0451 ERROR  | No dut intf "vlab-05|Ethernet28" in full_dut_fanout_port_map
> ```

Thanks @zjswhhh fixed. It shouldn't have run on `vs` testbed as there is no fanout here.

**Review by wangxin — APPROVED** [review]

**Pull request merged**